File: botgo/keyboards.go
package main

import (
	"fmt"
	"path/filepath"
	"strconv"
)

func adminKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "👥 Пользователи", CallbackData: "users:menu"}, {Text: "📝 Заявки", CallbackData: "users:requested:0"}},
		{{Text: "📣 Рассылка", CallbackData: "broadcast"}, {Text: "🔄 Синхр./восстановление", CallbackData: "maintenance"}},
		{{Text: "✏️ Тексты и кнопки", CallbackData: "content"}, {Text: "✨ Стикеры", CallbackData: "stickers"}},
	})
}

func backAdminKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{{{Text: "🛠️ В админку", CallbackData: "admin"}}})
}

func requestReviewKeyboard(id int64) *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{{{Text: "✅ Одобрить", CallbackData: fmt.Sprintf("approve:%d", id)}, {Text: "❌ Отклонить", CallbackData: fmt.Sprintf("reject:%d", id)}}})
}

func (a *App) accountKeyboard(lang string) *InlineKeyboardMarkup {
	suffix := "ru"
	if lang == "en" {
		suffix = "en"
	}
	cloud := a.content.Button("cloud_" + suffix)
	changePassword := a.content.Button("change_password_" + suffix)
	support := a.content.Button("support_" + suffix)
	donate := a.content.Button("donate_" + suffix)
	language := a.content.Button("language_" + suffix)
	rows := [][]InlineKeyboardButton{{{Text: cloud, URL: a.cfg.NextcloudURL}}, {{Text: changePassword, CallbackData: "account:change_password"}}}
	if a.cfg.EnableSupportBlock {
		rows = append(rows, []InlineKeyboardButton{{Text: support, CallbackData: "account:support"}})
	}
	if a.cfg.EnableDonateBlock {
		rows = append(rows, []InlineKeyboardButton{{Text: donate, CallbackData: "account:donate"}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: language, CallbackData: "account:language"}})
	return keyboard(rows)
}

func accountBackKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{{{Text: "⬅️ Назад", CallbackData: "account:home"}}})
}

func languageKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{{{Text: "🇷🇺 Русский", CallbackData: "lang:ru"}, {Text: "🇬🇧 English", CallbackData: "lang:en"}}, {{Text: "⬅️ Назад", CallbackData: "account:home"}}})
}

func donateKeyboard(cfg Config) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	if cfg.TelegramStarsEnabled && len(cfg.TelegramStarsAmounts) > 0 {
		rows = append(rows, []InlineKeyboardButton{{Text: "⭐ Telegram Stars", CallbackData: "donate:stars"}})
	}
	if cfg.PlategaVisible() {
		rows = append(rows, []InlineKeyboardButton{{Text: "💳 Platega", CallbackData: "donate:platega"}})
	}
	if cfg.DonateURL != "" {
		rows = append(rows, []InlineKeyboardButton{{Text: "💙 Донат", URL: cfg.DonateURL}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Назад", CallbackData: "account:home"}})
	return keyboard(rows)
}

func (cfg Config) PlategaVisible() bool {
	return cfg.PlategaEnabled && (cfg.PlategaPaymentURL() != "" || (cfg.PlategaMerchantID != "" && cfg.PlategaSecret != ""))
}

func (cfg Config) PlategaPaymentURL() string {
	if cfg.PlategaEnabled {
		return cfg.PlategaURL
	}
	return ""
}

func starsKeyboard(cfg Config) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	row := []InlineKeyboardButton{}
	for _, amount := range cfg.TelegramStarsAmounts {
		row = append(row, InlineKeyboardButton{Text: fmt.Sprintf("⭐ %d", amount), CallbackData: fmt.Sprintf("stars:%d", amount)})
		if len(row) == 3 {
			rows = append(rows, row)
			row = []InlineKeyboardButton{}
		}
	}
	if len(row) > 0 {
		rows = append(rows, row)
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Назад", CallbackData: "account:donate"}})
	return keyboard(rows)
}

func plategaKeyboard(cfg Config, apiEnabled bool) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	if apiEnabled {
		row := []InlineKeyboardButton{}
		for _, amount := range cfg.PlategaAmountsRUB {
			row = append(row, InlineKeyboardButton{Text: fmt.Sprintf("💳 %d RUB", amount), CallbackData: fmt.Sprintf("platega:%d", amount)})
			if len(row) == 2 {
				rows = append(rows, row)
				row = []InlineKeyboardButton{}
			}
		}
		if len(row) > 0 {
			rows = append(rows, row)
		}
	}
	if cfg.PlategaPaymentURL() != "" {
		rows = append(rows, []InlineKeyboardButton{{Text: "💳 Оплатить", URL: cfg.PlategaPaymentURL()}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Назад", CallbackData: "account:donate"}})
	return keyboard(rows)
}

func plategaPaymentKeyboard(paymentURL, transactionID string) *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "💳 Оплатить", URL: paymentURL}},
		{{Text: "🔎 Проверить оплату", CallbackData: "platega_check:" + transactionID}},
		{{Text: "⬅️ Назад", CallbackData: "donate:platega"}},
	})
}

func usersMenuKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "👥 Все", CallbackData: "users:all:0"}, {Text: "✅ Одобренные", CallbackData: "users:approved:0"}},
		{{Text: "📝 Заявки", CallbackData: "users:requested:0"}, {Text: "❌ Отклоненные", CallbackData: "users:rejected:0"}},
		{{Text: "🔎 Поиск", CallbackData: "users:search"}},
		{{Text: "🛠️ В админку", CallbackData: "admin"}},
	})
}

func usersKeyboard(users []User, status string, page int, hasNext bool) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	for _, user := range users {
		name := strPtr(user.Username, strPtr(user.FirstName, strconv.FormatInt(user.TelegramID, 10)))
		label := fmt.Sprintf("%s | %s | %dGB", name, user.Status, user.QuotaGB)
		if len([]rune(label)) > 60 {
			label = string([]rune(label)[:60])
		}
		rows = append(rows, []InlineKeyboardButton{{Text: label, CallbackData: fmt.Sprintf("user:%d:%s:%d", user.TelegramID, status, page)}})
	}
	nav := []InlineKeyboardButton{}
	if page > 0 {
		nav = append(nav, InlineKeyboardButton{Text: "⬅️ Назад", CallbackData: fmt.Sprintf("users:%s:%d", status, page-1)})
	}
	if hasNext {
		nav = append(nav, InlineKeyboardButton{Text: "➡️ Вперед", CallbackData: fmt.Sprintf("users:%s:%d", status, page+1)})
	}
	if len(nav) > 0 {
		rows = append(rows, nav)
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "🔎 Поиск", CallbackData: "users:search"}, {Text: "⬅️ Пользователи", CallbackData: "users:menu"}})
	return keyboard(rows)
}

func userKeyboard(user *User, backStatus string, backPage int) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	if user.Status == "requested" || user.Status == "rejected" {
		rows = append(rows, []InlineKeyboardButton{{Text: "✅ Одобрить", CallbackData: fmt.Sprintf("approve:%d", user.TelegramID)}})
	}
	if user.Status == "requested" {
		rows = append(rows, []InlineKeyboardButton{{Text: "❌ Отклонить", CallbackData: fmt.Sprintf("reject:%d", user.TelegramID)}})
	}
	if user.Status == "approved" {
		rows = append(rows, []InlineKeyboardButton{
			{Text: "➕ 1GB", CallbackData: fmt.Sprintf("quotaadd:%d:1", user.TelegramID)},
			{Text: "➕ 5GB", CallbackData: fmt.Sprintf("quotaadd:%d:5", user.TelegramID)},
			{Text: "➕ 10GB", CallbackData: fmt.Sprintf("quotaadd:%d:10", user.TelegramID)},
		})
		rows = append(rows, []InlineKeyboardButton{{Text: "⚙️ Другое", CallbackData: fmt.Sprintf("quotacustom:%d", user.TelegramID)}, {Text: "🔐 Сбросить пароль", CallbackData: fmt.Sprintf("resetpass:%d", user.TelegramID)}})
		if isPremium(user) {
			rows = append(rows, []InlineKeyboardButton{{Text: "⭐ Убрать премиум", CallbackData: fmt.Sprintf("supporter:%d:0", user.TelegramID)}})
		} else {
			rows = append(rows, []InlineKeyboardButton{{Text: "⭐ Сделать премиум", CallbackData: fmt.Sprintf("supporter:%d:1", user.TelegramID)}})
		}
		if user.IsDisabled == 1 {
			rows = append(rows, []InlineKeyboardButton{{Text: "🟢 Включить", CallbackData: fmt.Sprintf("enable:%d", user.TelegramID)}})
		} else {
			rows = append(rows, []InlineKeyboardButton{{Text: "🔴 Отключить", CallbackData: fmt.Sprintf("disable:%d", user.TelegramID)}})
		}
		rows = append(rows, []InlineKeyboardButton{{Text: "🗑️ Удалить", CallbackData: fmt.Sprintf("deleteask:%d", user.TelegramID)}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Назад", CallbackData: fmt.Sprintf("users:%s:%d", backStatus, backPage)}})
	return keyboard(rows)
}

func deleteConfirmKeyboard(id int64) *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{{{Text: "🗑️ Да, удалить", CallbackData: fmt.Sprintf("deleteyes:%d", id)}}, {{Text: "⬅️ Отмена", CallbackData: fmt.Sprintf("user:%d:all:0", id)}}})
}

func backupKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "🗄️ Создать PostgreSQL", CallbackData: "backup:db"}, {Text: "📦 Создать JSON", CallbackData: "backup:json"}},
		{{Text: "📋 Список", CallbackData: "backup:list"}, {Text: "♻️ Восстановить", CallbackData: "backup:restore"}},
		{{Text: "⬅️ Сервис", CallbackData: "maintenance"}},
	})
}

func maintenanceKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "🔄 Синхронизировать пользователей", CallbackData: "sync"}},
		{{Text: "☁️ Проверить мой клауд", CallbackData: "admincloud"}},
		{{Text: "🗄️ Бекапы и восстановление", CallbackData: "backup"}},
		{{Text: "🛠️ В админку", CallbackData: "admin"}},
	})
}

func contentKeyboard() *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "💬 Сообщения", CallbackData: "content:messages"}, {Text: "🔘 Кнопки", CallbackData: "content:buttons"}},
		{{Text: "🛠️ В админку", CallbackData: "admin"}},
	})
}

func contentListKeyboard(kind string) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	items := contentMessages
	if kind == "button" {
		items = contentButtons
	}
	for _, key := range sortedContentKeys(items) {
		rows = append(rows, []InlineKeyboardButton{{Text: contentTitle(items, key), CallbackData: "content:" + kind + ":" + key}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Тексты", CallbackData: "content"}})
	return keyboard(rows)
}

func contentEditKeyboard(kind, key string) *InlineKeyboardMarkup {
	return keyboard([][]InlineKeyboardButton{
		{{Text: "✏️ Изменить", CallbackData: "content:set:" + kind + ":" + key}},
		{{Text: "↩️ Сбросить", CallbackData: "content:reset:" + kind + ":" + key}},
		{{Text: "⬅️ Назад", CallbackData: "content:" + mapBool(kind == "message", "messages", "buttons")}},
	})
}

func stickersKeyboard(store *StickerStore, packURL string) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	for _, event := range stickerEvents {
		mark := eventMark(event)
		if value, ok := store.Get(event); ok {
			if value.Kind == StickerKindCustomEmoji {
				mark = "🧩"
			} else {
				mark = "🖼️"
			}
		}
		rows = append(rows, []InlineKeyboardButton{{Text: mark + " " + event, CallbackData: "sticker:event:" + event}})
	}
	if packURL != "" {
		rows = append(rows, []InlineKeyboardButton{{Text: "🧪 Открыть CPT_Emoji", URL: packURL}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "🛠️ В админку", CallbackData: "admin"}})
	return keyboard(rows)
}

func stickerEventKeyboard(event string, hasValue bool, packURL string) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{
		{{Text: "➕ Установить", CallbackData: "sticker:set:" + event}},
	}
	if hasValue {
		rows = append(rows, []InlineKeyboardButton{
			{Text: "👁️ Предпросмотр", CallbackData: "sticker:preview:" + event},
			{Text: "🧹 Очистить", CallbackData: "sticker:clear:" + event},
		})
	}
	if packURL != "" {
		rows = append(rows, []InlineKeyboardButton{{Text: "🧪 Открыть CPT_Emoji", URL: packURL}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Стикеры", CallbackData: "stickers"}})
	return keyboard(rows)
}

func restoreBackupKeyboard(files []string) *InlineKeyboardMarkup {
	rows := [][]InlineKeyboardButton{}
	for index, file := range files {
		label := filepath.Base(file)
		if len([]rune(label)) > 60 {
			label = string([]rune(label)[:60])
		}
		rows = append(rows, []InlineKeyboardButton{{Text: label, CallbackData: fmt.Sprintf("restore:%d", index)}})
	}
	rows = append(rows, []InlineKeyboardButton{{Text: "⬅️ Отмена", CallbackData: "backup"}})
	return keyboard(rows)
}

func keyboard(rows [][]InlineKeyboardButton) *InlineKeyboardMarkup {
	return &InlineKeyboardMarkup{InlineKeyboard: rows}
}

File: bot/keyboards.py
from __future__ import annotations

from aiogram.types import InlineKeyboardMarkup
from aiogram.utils.keyboard import InlineKeyboardBuilder


def _labels(lang: str) -> dict[str, str]:
    if lang == "en":
        return {
            "change_password": "🔐 Change password",
            "cloud_login": "☁️ Open cloud",
            "support": "💬 Support",
            "donate": "💙 Donate",
            "stars": "⭐ Telegram Stars",
            "platega": "💳 Platega",
            "pay": "💳 Pay",
            "check_payment": "🔎 Check payment",
            "language": "🌐 Language",
            "back": "⬅️ Back",
            "ru": "Русский",
            "en": "English",
        }
    return {
        "change_password": "🔐 Сменить пароль",
        "cloud_login": "☁️ Войти в облако",
        "support": "💬 Поддержка",
        "donate": "💙 Донат",
        "stars": "⭐ Telegram Stars",
        "platega": "💳 Platega",
        "pay": "💳 Оплатить",
        "check_payment": "🔎 Проверить оплату",
        "language": "🌐 Язык",
        "back": "⬅️ Назад",
        "ru": "Русский",
        "en": "English",
    }


def account_keyboard(
    lang: str = "ru",
    show_support: bool = True,
    show_donate: bool = True,
    cloud_url: str | None = None,
) -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    if cloud_url:
        builder.button(text=labels["cloud_login"], url=cloud_url)
    builder.button(text=labels["change_password"], callback_data="account:change_password")
    if show_support:
        builder.button(text=labels["support"], callback_data="account:support")
    if show_donate:
        builder.button(text=labels["donate"], callback_data="account:donate")
    builder.button(text=labels["language"], callback_data="account:language")
    builder.adjust(1)
    return builder.as_markup()


def account_back_keyboard(lang: str = "ru") -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    builder.button(text=labels["back"], callback_data="account:home")
    builder.adjust(1)
    return builder.as_markup()


def support_keyboard(lang: str = "ru") -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    builder.button(text=labels["back"], callback_data="account:home")
    builder.adjust(1)
    return builder.as_markup()


def donate_keyboard(
    lang: str = "ru",
    show_stars: bool = False,
    show_platega: bool = False,
    donate_url: str | None = None,
) -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    if show_stars:
        builder.button(text=labels["stars"], callback_data="donate:stars")
    if show_platega:
        builder.button(text=labels["platega"], callback_data="donate:platega")
    if donate_url:
        builder.button(text=labels["donate"], url=donate_url)
    builder.button(text=labels["back"], callback_data="account:home")
    builder.adjust(1)
    return builder.as_markup()


def stars_amounts_keyboard(lang: str = "ru", amounts: tuple[int, ...] = ()) -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    for amount in amounts:
        builder.button(text=f"⭐ {amount}", callback_data=f"stars:{amount}")
    builder.button(text=labels["back"], callback_data="account:donate")
    builder.adjust(3, 1)
    return builder.as_markup()


def platega_amounts_keyboard(
    lang: str = "ru",
    amounts: tuple[int, ...] = (),
    static_url: str | None = None,
) -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    for amount in amounts:
        builder.button(text=f"💳 {amount} RUB", callback_data=f"platega:{amount}")
    if static_url:
        builder.button(text=labels["pay"], url=static_url)
    builder.button(text=labels["back"], callback_data="account:donate")
    builder.adjust(2, 1, 1)
    return builder.as_markup()


def platega_payment_keyboard(lang: str, payment_url: str, transaction_id: str) -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    builder.button(text=labels["pay"], url=payment_url)
    builder.button(text=labels["check_payment"], callback_data=f"platega_check:{transaction_id}")
    builder.button(text=labels["back"], callback_data="donate:platega")
    builder.adjust(1)
    return builder.as_markup()


def language_keyboard(lang: str = "ru") -> InlineKeyboardMarkup:
    labels = _labels(lang)
    builder = InlineKeyboardBuilder()
    builder.button(text=f"🇷🇺 {labels['ru']}", callback_data="lang:ru")
    builder.button(text=f"🇬🇧 {labels['en']}", callback_data="lang:en")
    builder.button(text=labels["back"], callback_data="account:home")
    builder.adjust(2, 1)
    return builder.as_markup()


def request_review_keyboard(telegram_id: int) -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    builder.button(text="✅ Одобрить", callback_data=f"approve:{telegram_id}")
    builder.button(text="❌ Отклонить", callback_data=f"reject:{telegram_id}")
    builder.adjust(2)
    return builder.as_markup()


def admin_keyboard() -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    builder.button(text="👥 Пользователи", callback_data="users:all:0")
    builder.button(text="📝 Заявки", callback_data="users:requested:0")
    builder.button(text="🗄️ Бекапы", callback_data="backup")
    builder.button(text="📣 Рассылка", callback_data="broadcast")
    builder.button(text="🔄 Синхронизация", callback_data="sync")
    builder.button(text="✨ Стикеры", callback_data="stickers")
    builder.adjust(2, 2, 1)
    return builder.as_markup()


def users_keyboard(users: list[dict], status: str, page: int, has_next: bool) -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    for user in users:
        name = user.get("username") or user.get("first_name") or str(user["telegram_id"])
        label = f"{name} | {user['status']} | {user.get('quota_gb', 0)}GB"
        builder.button(text=label[:60], callback_data=f"user:{user['telegram_id']}:{status}:{page}")

    nav_buttons = []
    if page > 0:
        nav_buttons.append(("⬅️ Назад", f"users:{status}:{page - 1}"))
    if has_next:
        nav_buttons.append(("➡️ Вперед", f"users:{status}:{page + 1}"))
    for text, callback_data in nav_buttons:
        builder.button(text=text, callback_data=callback_data)

    builder.button(text="🛠️ В админку", callback_data="admin")
    builder.adjust(1)
    return builder.as_markup()


def user_keyboard(
    telegram_id: int,
    back_status: str,
    back_page: int,
    status: str,
    is_disabled: bool,
    is_supporter: bool,
) -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    if status in {"requested", "rejected"}:
        builder.button(text="✅ Одобрить", callback_data=f"approve:{telegram_id}")
    if status == "requested":
        builder.button(text="❌ Отклонить", callback_data=f"reject:{telegram_id}")
    if status == "approved":
        builder.button(text="➕ 1GB", callback_data=f"quotaadd:{telegram_id}:1")
        builder.button(text="➕ 5GB", callback_data=f"quotaadd:{telegram_id}:5")
        builder.button(text="➕ 10GB", callback_data=f"quotaadd:{telegram_id}:10")
        builder.button(text="⚙️ Другое", callback_data=f"quotacustom:{telegram_id}")
        builder.button(text="🔐 Сбросить пароль", callback_data=f"resetpass:{telegram_id}")
        if is_supporter:
            builder.button(text="⭐ Убрать премиум", callback_data=f"supporter:{telegram_id}:0")
        else:
            builder.button(text="⭐ Сделать премиум", callback_data=f"supporter:{telegram_id}:1")
        if is_disabled:
            builder.button(text="🟢 Включить", callback_data=f"enable:{telegram_id}")
        else:
            builder.button(text="🔴 Отключить", callback_data=f"disable:{telegram_id}")
        builder.button(text="🗑️ Удалить", callback_data=f"deleteask:{telegram_id}")
    builder.button(text="⬅️ Назад", callback_data=f"users:{back_status}:{back_page}")
    builder.adjust(2, 3, 1, 1, 1, 1)
    return builder.as_markup()


def delete_confirm_keyboard(telegram_id: int) -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    builder.button(text="🗑️ Да, удалить", callback_data=f"deleteyes:{telegram_id}")
    builder.button(text="⬅️ Отмена", callback_data=f"user:{telegram_id}:all:0")
    builder.adjust(1)
    return builder.as_markup()


def backup_keyboard() -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    builder.button(text="🗄️ Создать SQLite", callback_data="backup:db")
    builder.button(text="📦 Создать JSON", callback_data="backup:json")
    builder.button(text="📋 Список", callback_data="backup:list")
    builder.button(text="♻️ Восстановить", callback_data="backup:restore")
    builder.button(text="🛠️ В админку", callback_data="admin")
    builder.adjust(2, 2, 1)
    return builder.as_markup()


def restore_backup_keyboard(backups: list[tuple[str, str]]) -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    for backup_id, label in backups:
        builder.button(text=label[:60], callback_data=f"restore:{backup_id}")
    builder.button(text="⬅️ Отмена", callback_data="backup")
    builder.adjust(1)
    return builder.as_markup()


def broadcast_confirm_keyboard() -> InlineKeyboardMarkup:
    builder = InlineKeyboardBuilder()
    builder.button(text="📣 Отправить всем", callback_data="broadcast:confirm")
    builder.button(text="⬅️ Отмена", callback_data="broadcast:cancel")
    builder.adjust(1)
    return builder.as_markup()
